src/Entities/Pawn.cpp

```diff
@@ -111,6 +111,37 @@ void cPawn::Tick(std::chrono::milliseconds a_Dt, cChunk & a_Chunk)
 void cPawn::KilledBy(TakeDamageInfo & a_TDI)
 {
 	ClearEntityEffects();
+
+	if (IsPlayer())
+	{
+		auto Player = static_cast<cPlayer *>(this);
+		cInventory & inv = Player->GetInventory();
+
+		//  If the player is holding a totem of undying in their off-hand or
+		//  main-hand slot and receives otherwise fatal damage, the totem saves the player from death.
+		if ((inv.GetShieldSlot().m_ItemType == E_ITEM_TOTEM_OF_UNDYING ||
+			 inv.GetEquippedItem().m_ItemType == E_ITEM_TOTEM_OF_UNDYING) &&
+				a_TDI.DamageType != dtAdmin && a_TDI.DamageType != dtInVoid
+			)
+		{
+			inv.SetShieldSlot(cItem());
```

Comment:
Won't this clear the shield slot, even if the totem was in the hotbar?

Reply:
Sleep deprivation... I fixed it in a new commit.